### DISCUSSION TITLE
what is the best way to do localhost (in my machine) visual testing?

### DISCUSSION THREAD

**Discussion by ChenReuven** [storybookjs/storybook#22868] [Help]
as far as I see from this discussion, Storyshots is quite "Dead" and not maintenance.
[](https://github.com/storybookjs/storybook/discussions/18119)

I try to figure out what is the best way to do a visual test in my localhost machine via test-runner?

Thanks

**Comment by squidjam** [comment]
This page has the recipe: https://storybook.js.org/addons/@storybook/test-runner#:~:text=%2C%0A%7D%3B-,Image%20snapshot%20recipe,-Here%27s%20a%20slightly

I suggest you bookmark this one as it has other interesting things that (for me) are difficult to find elsewhere. 
I sure hope this was the test-runner manual page.
